src/messages.rs

```diff
@@ -319,15 +274,16 @@ impl<CS: CipherSuite> CredentialResponse<CS> {
     pub fn serialize(&self) -> Vec<u8> {
         let mut credential_response: Vec<u8> = Vec::new();
         credential_response.extend_from_slice(&serialize(&self.beta.to_arr(), 2));
+        credential_response.extend_from_slice(&serialize(&self.server_s_pk.to_arr().to_vec(), 2));
         credential_response.extend_from_slice(&self.envelope.to_bytes());
         credential_response.extend_from_slice(&self.ke2_message.to_bytes());
         credential_response
     }
 
     /// Deserialization from bytes
     pub fn deserialize(input: &[u8]) -> Result<Self, ProtocolError> {
-        let (beta_bytes, envelope_and_ke2m_bytes) = tokenize(&input, 2)?;
-        let concatenated = [&beta_bytes[..], &envelope_and_ke2m_bytes[..]].concat();
+        let (beta_bytes, remainder) = tokenize(&input, 2)?;
+        let concatenated = [&beta_bytes[..], &remainder[..]].concat();
```

Comment:
nit: I get we have more cases to cover, but can we still find a more descriptive name than "remainder"?